Milvus.Client/MilvusCollection.Entity.cs

```diff
@@ -418,61 +418,144 @@ public Task<FlushResult> FlushAsync(CancellationToken cancellationToken = defaul
+        }
+
+        var describeResponse = await _client.InvokeAsync(
+                _client.GrpcClient.DescribeCollectionAsync,
+                new DescribeCollectionRequest { CollectionName = Name },
+                r => r.Status,
+                cancellationToken)
+                .ConfigureAwait(false);
+
+        var pkField = describeResponse.Schema.Fields.FirstOrDefault(x => x.IsPrimaryKey);
+        if (pkField == null)
+        {
+            throw new MilvusException("Schema must contain pk field");
+        }
+
+        var isUserRequestPkField = parameters?.OutputFieldsInternal?.Contains(pkField.Name) ?? false;
+        var userExpression = expression;
+        var userLimit = parameters?.Limit ?? int.MaxValue;
+
+        QueryRequest request = new()
+        {
+            CollectionName = Name,
+            Expr = (userExpression, pkField) switch
             {
-                request.TravelTimestamp = parameters.TimeTravelTimestamp.Value;
+                // If user expression is not null, we should use it
+                {userExpression: not null} => userExpression,
+                // If user expression is null and pk field is string
+                {pkField.DataType: DataType.VarChar} => $"{pkField.Name} != ''",
+                // If user expression is null and pk field is not string
+                _ => $"{pkField.Name} < {long.MaxValue}",
             }
+        };
 
-            if (parameters.PartitionNamesInternal?.Count > 0)
+        PopulateQueryRequestFromParameters(request, parameters);
+
+        // Request id field in any case to proceed with an iterations
+        if (!isUserRequestPkField) request.OutputFields.Add(pkField.Name);
+
+        // Replace parameters required for iterator
+        ReplaceKeyValueItems(request.QueryParams,
+            new Grpc.KeyValuePair {Key = Constants.Iterator, Value = "True"},
+            new Grpc.KeyValuePair {Key = Constants.ReduceStopForBest, Value = "True"},
+            new Grpc.KeyValuePair {Key = Constants.BatchSize, Value = batchSize.ToString(CultureInfo.InvariantCulture)},
+            new Grpc.KeyValuePair {Key = Constants.Offset, Value = 0.ToString(CultureInfo.InvariantCulture)},
+            new Grpc.KeyValuePair {Key = Constants.Limit, Value = Math.Min(batchSize, userLimit).ToString(CultureInfo.InvariantCulture)});
+
+        var processedItemsCount = 0;
+        while (!cancellationToken.IsCancellationRequested)
+        {
+            var response = await _client.InvokeAsync(
+                    _client.GrpcClient.QueryAsync,
+                    request,
+                    static r => r.Status,
+                    cancellationToken)
+                .ConfigureAwait(false);
+
+            object? pkLastValue;
+            int processedDuringIterationCount;
+            var pkFieldsData = response.FieldsData.Single(x => x.FieldId == pkField.FieldID);
+            if (pkField.DataType == DataType.VarChar)
```

Comment:
It is direct port of logic from python client

https://github.com/milvus-io/pymilvus/blob/6625af7fd0bfdf75fd596e7a9a871528cb539ef7/pymilvus/orm/iterator.py#L186 

But ok, will add

Reply:
In general, I don't think it makes sense to keep code a certain way just because the python implementation looks a certain way. This is a different project in a different language, and we should be doing things in the best way possible here regardless of what the Python driver looks like.

Reply:
Added